### DISCUSSION TITLE
Manually triggering recurring transactions

### DISCUSSION THREAD

**Discussion by niklas152** [firefly-iii/firefly-iii#6215] [Firefly III]
Hello,

I would love to have the ability to manually trigger set up recurring transactions.
I imagine it would be similar (from an UI viewpoint) as manually triggering rules. Manually triggering an recurring transaction would create the transaction just as if the date at which it would have executed automatically was reached, thus moving the next automatic execution one <time unit of specific transaction> further out, essentially moving the regular next execution to now but leaving future executions alone (or it could ask whether to actually move the next one, allowing to "duplicate" execution?)
An possible use case for this is, e.g., having some distribution of money to different (firefly) accounts after receiving your salary. Now you can do this using recurring transaction already, however, when the exact date of when you receive your salary fluctates, you run into the problem of either needing to set the recurring transactions to a very late date, such that the probability of having received your salary until then is high but with the drawback that when it comes earlier the money is sitting unused until then or setting it earlier with the risk that money gets distributed which isn't there yet. With the ability to manually trigger them, you could go the former route but instead of having it sit around pointlessly when it does come earlier, you can now issue the transactions manually as soon as it arrives.

In combination, it would also be nice to be able to group recurring transactions similarly as we can now group rules, analogously allowing to manually trigger a whole group of recurring transactions at once.

**Comment by JC5** [comment]
Both of these things are on my list, yes :+1:
